internal/pkg/testproject/snapshot.go

```diff
@@ -0,0 +1,123 @@
+package testproject
+
+import (
+	"strings"
+
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/client"
+	"github.com/keboola/keboola-as-code/internal/pkg/fixtures"
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+// NewSnapshot - to validate final project state in tests.
+func (p *Project) NewSnapshot() (*fixtures.ProjectSnapshot, error) {
+	snapshot := &fixtures.ProjectSnapshot{}
+	configs := make(map[string]*fixtures.Config)
+	var schedules []*model.Schedule
+
+	// Load Storage API object and Schedules in parallel
+	grp := &errgroup.Group{}
+	grp.Go(func() error {
+		return p.snapshot(snapshot, configs)
+	})
+	grp.Go(func() (err error) {
+		schedules, err = p.SchedulerApi().ListSchedules()
+		return err
+	})
+	if err := grp.Wait(); err != nil {
+		return nil, err
+	}
+
+	// Map schedules
+	for _, schedule := range schedules {
+		configKey := model.ConfigKey{BranchId: p.DefaultBranch().Id, ComponentId: model.SchedulerComponentId, Id: schedule.ConfigurationId}
+		scheduleConfig := configs[configKey.String()]
+		snapshot.Schedules = append(snapshot.Schedules, &fixtures.Schedule{Name: scheduleConfig.Name})
+	}
+
+	// Sort by name
+	utils.SortByName(snapshot.Branches)
+	for _, b := range snapshot.Branches {
+		utils.SortByName(b.Configs)
+		for _, c := range b.Configs {
+			utils.SortByName(c.Rows)
+		}
+	}
+
+	return snapshot, nil
+}
+
+func (p *Project) snapshot(snapshot *fixtures.ProjectSnapshot, configs map[string]*fixtures.Config) error {
+	// Load objects from Storage API
+	// Branches
+	pool := p.StorageApi().NewPool()
```

Comment:
Musel som zrefactorovat generovanie `snapshot` aby sa pouzivalo priamo API a nie state -> ako sme uz rozpravali.

Ide o to, ze `State` mi odstrani z konfiguracie `relations`, napr. `variables_id` ... a neviem tak napisat test pre `push-variables`, ci sa v API vsetko spravne ulozilo.